### ISSUE TITLE
Problems with configuring Jupyter start directory

### ISSUE THREAD

**Issue by ijstokes** [ContinuumIO/anaconda-issues#1730]
Taken from the Anaconda ML, posted by (I think) @hraftery, discussing the problems he's having on Windows getting Jupyter, via Anaconda, to use the right start directory (launched from Navigator or via the Start menu):

The idea of changing the default directory that appears when Jupyter Notebook launches in the browser has been covered many times, most usefully here:

https://groups.google.com/a/continuum.io/forum/#!topic/anaconda/gqRwT_SxGBw

and here:

https://stackoverflow.com/questions/35254852/how-to-change-jupyter-start-folder/43213658

The generally accepted solution is to define the c.NotebookApp.notebook_dir setting in jupyter_notebook_config.py exactly right (forward slashes, spaces double quoted). This works just fine on Windows 10 when starting Jupyter Notebook from the command line or from Anaconda Navigator.

The most convenient way I find to start Jupyter Notebook however, is with the shortcut "Jupyter Notebook" that is installed in the Start Menu. Alas, while the c.NotebookApp.notebook_dir setting is certainly read (because setting it incorrectly causes an error), it is ignored, and the default directory stubbornly stays as my home directory.

Ideally, this shortcut would use the c.NotebookApp.notebook_dir setting (so you only have to set it once, the change can be added to configuration management). But as a workaround I suspect changing the "Start in" directory of the shortcut will make a difference. Alas, as described here:

https://groups.google.com/a/continuum.io/forum/#!topic/anaconda/7JrSKLVBTPo

Windows can't handle shortcuts with a target longer than 256 characters. Attempting to change the "Start In" directory truncates the "Target" setting, leaving the shortcut irreparably broken (I've now re-installed Anaconda more than a dozen times trying to get to the bottom of this).

So in conclusion, can the "Jupyter Notebook" shortcut be fixed to observe the c.NotebookApp.notebook_dir setting correctly? Or as a non-ideal workaround, is there a way to change the default directory used by the shortcut, without destroying the shortcut itself?

**Comment by ijstokes** [comment]
@hraftery then provided a work-around to his problem:

I now have a clunky workaround to my own question. Predictably enough, it's the special kind of ugly - Windows ugly.

Turns out changing the "Start in" field has no effect. The trigger is the last argument of the target, which by default is %USERPROFILE%. For some reason I can't determine, that sets the notebook directory. To modify, create this delightful VBScript file, and call it something like "modify_shortcut.vbs":

```
Const OrigShortcut = "C:\path\to\existing\Jupyter Notebook.lnk"
Const NewShortcut = "C:\path\to\new\Jupyter Notebook - modified.lnk"
Const SACDir = "C:\path\to\notebook\dir"

' Copy original shortcut
Set fso = CreateObject("Scripting.FileSystemObject")
fso.CopyFile OrigShortcut, NewShortcut, True
Set fso = Nothing

' Modify shortcut
Set sh = CreateObject("WScript.Shell")
Set shortcut = sh.CreateShortcut(NewShortcut)
args = shortcut.Arguments
shortcut.Arguments = Replace(args, "%USERPROFILE%", """" & SACDir & """")
shortcut.Save
```

Modify the 3 Const variables to suit, then save. Either double click the resulting vbs file or run from the command line using cscript modify_shortcut.vbs. Note the NewShortcut file is created. Double-click it and be finally blessed with the correct notebook dir!

**Comment by stumitchell** [comment]
Also I have just checked and if you remove the %USERPROFILE% you will get the notebook started in the dir that is defined in c.NotebookApp.notebook_dir

**Comment by SirDijkhoff** [comment]
Hi all, 

how do you remove the %USERPROFILE% ?

**Comment by stumitchell** [comment]
remove it from the properties window of your desktop shortcut
